keras/layers/rnn/base_rnn.py

```diff
@@ -905,8 +905,22 @@ def reset_states(self, states=None):
                         self.variable_dtype or backend.floatx(),
                     )
                 )
+            if len(flat_init_state_values) == 1:
+
+                def assign_unique_var(x):
+                    return backend.variable(x, name="state")
+
+            else:
+                unique_id = 0
+
+                def assign_unique_var(x):
+                    nonlocal unique_id
```

Comment:
I think calling them state_0/1 is also bit weird, since the states in LSTM actually carry meaningful information (hidden state and output). So if  you could like to properly name the variable, then let's do that instead.

Also if you have 2 stateful lstm (eg bi-directional), what will be the state variable names? (state_0/1 and state_0/1 or state_0/1/2/3)?